src/dbnode/client/fetch_state.go

```diff
@@ -113,15 +143,23 @@ func (f *fetchState) completionFn(
 		return
 	}
 
-	opts, ok := result.(fetchTaggedResultAccumulatorOpts)
-	if !ok {
+	var (
+		done bool
+		err  error
+	)
+	switch r := result.(type) {
+	case fetchTaggedResultAccumulatorOpts:
```

Comment:
should we also test that `f.stateType == fetchTaggedFetchState` here (just paranoia)